File: app/src/main/java/com/yagi/android/numbercatch/FinishActivity.java
package com.yagi.android.numbercatch;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.Html;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

public class FinishActivity extends AppCompatActivity{

    int score;
    TextView finalScoreText;
    TextView bestScoreText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.layout_game_finish);

        Intent i = getIntent();
        score = i.getIntExtra("score", 0);
        finalScoreText = (TextView)findViewById(R.id.textView_FinalScore);
        bestScoreText = (TextView)findViewById(R.id.textView_bestScore);
        Log.d("intent2", String.valueOf(score));
        finalScoreText.setText(String.valueOf(score));
        Log.d("text", String.valueOf(finalScoreText));
        SharedPreferences mHighScorePref = this.getSharedPreferences("NumberCatch", 0);
        int highScore = mHighScorePref.getInt("BEST SCORE", 0);
        if (highScore < this.score) {
            SharedPreferences.Editor editor = mHighScorePref.edit();
            editor.putInt("BEST SCORE", this.score);
            editor.apply();
            this.bestScoreText.setText(String.valueOf(this.score));
        } else {
            this.bestScoreText.setText(String.valueOf(highScore));
        }
    }

    public void restart (View v){
        Intent i = new Intent(this, MainActivity.class);
        startActivity(i);
    }


}

File: app/src/main/java/com/yagi/android/numbercatch/FinishActivity.kt
package com.yagi.android.numbercatch

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class FinishActivity : AppCompatActivity() {
    var score = 0
    var finalScoreText: TextView? = null
    var bestScoreText: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.layout_game_finish)
        val i = intent
        score = i.getIntExtra("score", 0)
        finalScoreText = findViewById<View>(R.id.textView_FinalScore) as TextView
        bestScoreText = findViewById<View>(R.id.textView_bestScore) as TextView
        finalScoreText!!.text = score.toString()
        val mHighScorePref = getSharedPreferences("NumberCatch", 0)
        val highScore = mHighScorePref.getInt("BEST SCORE", 0)
        if (highScore < score) {
            val editor = mHighScorePref.edit()
            editor.putInt("BEST SCORE", score)
            editor.apply()
            bestScoreText!!.text = score.toString()
        } else {
            bestScoreText!!.text = highScore.toString()
        }
    }

    fun restart() {
        val i = Intent(this, MainActivity::class.java)
        startActivity(i)
        finish()
    }
}